src/workerd/io/worker-entrypoint.c++

```diff
@@ -764,12 +792,21 @@ kj::Promise<WorkerInterface::CustomEvent::Result> WorkerEntrypoint::customEvent(
   auto promise = event->run(kj::mv(incomingRequest), entrypointName, kj::mv(props), waitUntilTasks)
                      .attach(kj::mv(event));
 
+  /*KJ_DEFER({
+    KJ_IF_SOME(t, context.getWorkerTracer()) {
+      t.setReturn(kj::none, context.now());
+    }
+  });*/
   // TODO(cleanup): In theory `context` may have been destroyed by now if `event->run()` dropped
   //   the `incomingRequest` synchronously. No current implementation does that, and
   //   maybeAddGcPassForTest() is a no-op outside of tests, so I'm ignoring the theoretical problem
   //   for now. Otherwise we will need to `atomicAddRef()` the `Worker` at some point earlier on
   //   but I'd like to avoid that in the non-test case.
-  return maybeAddGcPassForTest(context, kj::mv(promise));
+  auto result = co_await maybeAddGcPassForTest(context, kj::mv(promise));
+  KJ_IF_SOME(t, context.getWorkerTracer()) {
```

Comment:
Somehow, downstream tests segfault in the context.getWorkerTracer() call when run on CI, but pass when running locally

Reply:
Well it would run right after the first async operation in event->run which is after the onset event was submitted.